### PR TITLE
Removed one of the duplicate '/_nodes/{metric}' and '/_nodes/{node_id}' paths.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Was trying to load the spec with the Ruby openapi3_parser and it failed with this error.

```
/Users/dblock/.rvm/gems/ruby-3.1.2/gems/openapi3_parser-0.9.2/lib/openapi3_parser/node_factory/map.rb:206:in `validate': Invalid data for #/paths: There are paths that conflict: '/_nodes/{metric}', '/_nodes/{node_id}' (Openapi3Parser::Error::InvalidData)
```

Now properly detecting dups.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by nhtruong** [comment]
This will cause a breaking change in the clients. The clients accepts both `metric` and `node_id` as params for `client.nodes.info()` function ([js](https://github.com/opensearch-project/opensearch-js/blob/main/api/api/nodes.js), [ruby](https://github.com/opensearch-project/opensearch-ruby/blob/main/lib/opensearch/api/actions/nodes/info.rb)) and uses the following logic to determine which path to use:
- both `metric` and `node_id` are provided: `/nodes/{node_id}/{metric}`
- only `node_id` is provided: `/nodes/{node_id}`
- only `metric` is provided: `/nodes/{metric}`
- neither is provided: `/nodes`
This is another quirk of OpenSearch API.
The server treats`/nodes/{node_id}` and  `/nodes/{metric}` as an identical endpoint. So this endpoint is actually `/nodes/{node_id_or_metric}`. But this will cause the clients to read `node_id_or_metric` as the 3rd path param. We can collapse those 2 endpoints into `/nodes/{node_id_or_metric}` but we will have to add a new `x-` extension to tell the generator how to handle this edge case.

File: tools/tests/linter/fixtures/folder_validators/namespaces/invalid_folder/dup_path_d.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+paths:
+  '/nodes/{metric}': {}
+  '/nodes/{node_id}': {}
```
Comment (nhtruong) [inline tools/tests/linter/fixtures/folder_validators/namespaces/invalid_folder/dup_path_d.yaml:3]: We should also account for when the paths appear in different namespace files.

Comment (dblock) [reply tools/tests/linter/fixtures/folder_validators/namespaces/invalid_folder/dup_path_d.yaml:3]: https://github.com/opensearch-project/opensearch-api-specification/pull/416/commits/b3da76c27c337fce57973102e77e65eb8ed75635

**Comment by dblock** [comment]
> I'm thinking `x-union-param: ['node_id', 'metric']` for the `node_id_or_metric` path param

I made the change by introducing `/_nodes/{node_id_or_metric}`.

I am not sure we're doing the right thing by introducing parameter hints. As written in this PR the parameter is `node_id_or_metric`, and it can now be an array of either a `node_id` or a `metric` (`anyOf`). Perhaps, the right way is for  clients to patch this behavior to be backwards compatible instead of the spec requiring a hint?

File: spec/namespaces/nodes.yaml [hunk 2]
```diff
@@ -545,15 +530,21 @@ components:
...
+    nodes.info::path.node_id_or_metric:
       in: path
-      name: metric
-      description: Limits the information returned to the specific metrics. Supports a comma-separated list, such as http,ingest.
+      name: node_id_or_metric
+      description: |
+        Limits the information returned to a list of node IDs or specific metrics.
+        Supports a comma-separated list, such as node1,node2 or http,ingest.
       required: true
       schema:
```
Comment (Xtansia) [inline spec/namespaces/nodes.yaml:540]: NodeIds is already an array, so need to lift up the anyOf
```suggestion
      schema:
        anyOf:
          - title: node_id
            $ref: '../schemas/_common.yaml#/components/schemas/NodeIds'
          - title: metric
            type: array
            items:
              $ref: '../schemas/nodes.info.yaml#/components/schemas/Metric'
```

**Review by Xtansia — CHANGES_REQUESTED** [review]

**Comment by dblock** [comment]
> We should update https://github.com/opensearch-project/opensearch-api-specification/blob/main/CLIENT_GENERATOR_GUIDE.md to alert devs about this edge case as well.

Done, https://github.com/opensearch-project/opensearch-api-specification/pull/416/commits/d4f3a55c9fd8cab683faa1c30416e25936724a39.

**Review by nhtruong — APPROVED** [review]

File: spec/namespaces/nodes.yaml [hunk 3]
```diff
@@ -545,15 +530,28 @@ components:
...
+            items:
+              $ref: '../schemas/nodes.info.yaml#/components/schemas/Metric'
         type: array
         items:
-          $ref: '../schemas/nodes.info.yaml#/components/schemas/Metric'
+          anyOf:
+            - $ref: '../schemas/_common.yaml#/components/schemas/NodeIds'
+              title: node_id
+            - $ref: '../schemas/nodes.info.yaml#/components/schemas/Metric'
+              title: metric
```
Comment (Xtansia) [inline spec/namespaces/nodes.yaml:554]: Need to remove these duplicated parts

**Review by Xtansia — CHANGES_REQUESTED** [review]

File: spec/namespaces/nodes.yaml (hunk 3 above)
Comment (dblock) [reply spec/namespaces/nodes.yaml]: 🤦

**Review by Xtansia — APPROVED** [review]
